src/map.cpp

```diff
@@ -238,95 +237,96 @@ bool Map::placeCreature(const Position& centerPos, Creature* creature, bool exte
 
 	int32_t index = 0;
 	uint32_t flags = 0;
-	Item* toItem = nullptr;
+	std::shared_ptr<Item> toItem = nullptr;
 
-	Cylinder* toCylinder = tile->queryDestination(index, *creature, &toItem, flags);
-	toCylinder->internalAddThing(creature);
+	auto toThing = tile->queryDestination(index, creature, toItem, flags);
```

Comment:
Could this be `const auto`
```suggestion
	const auto toThing = tile->queryDestination(index, creature, toItem, flags);
```